[mqtt.homeassistant] Integration tests fail

Build on Windows 10:
```
[INFO] --- bnd-testing-maven-plugin:4.2.0:testing (default) @ org.openhab.binding.mqtt.homeassistant.tests ---
[INFO] Matching glob *
08:11:59.785 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created thread pool 'safeCall' with size 5
08:11:59.848 [main] DEBUG org.eclipse.smarthome.core.internal.i18n.I18nProviderImpl - No language set, setting locale to 'null'.
08:11:59.848 [main] DEBUG org.eclipse.smarthome.core.internal.i18n.I18nProviderImpl - Error setting measurement system for value 'null'.
08:11:59.895 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'scheduler' of size 5
08:11:59.926 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'file-processing' of size 5
08:12:00.004 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created thread pool 'ConfigStatusService' with size 5
08:12:00.035 [OH-file-processing-4] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/binding/binding.xml' in module 'org.openhab.binding.mqtt'...
08:12:00.051 [OH-file-processing-2] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/homeassistant-channel-config.xml' in module 'org.openhab.binding.mqtt.homeassistant'...
08:12:00.051 [OH-file-processing-3] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/i18n.xml' in module 'org.openhab.core'...
08:12:00.051 [OH-file-processing-1] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/config.xml' in module 'org.openhab.core.config.discovery'...
08:12:00.051 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/number-channel-config.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.051 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'discovery' of size 5
08:12:00.066 [main] DEBUG org.eclipse.smarthome.config.discovery.AbstractDiscoveryService - Background discovery for discovery service 'org.openhab.binding.mqtt.homeassistant.internal.discovery.HomeAssistantDiscovery' enabled.
08:12:00.066 [main] DEBUG org.eclipse.smarthome.config.discovery.AbstractDiscoveryService - Background discovery for discovery service 'org.openhab.binding.mqtt.internal.discovery.MqttServiceDiscoveryService' enabled.
08:12:00.066 [OH-file-processing-4] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/brokerConnectionInstance.xml' in module 'org.openhab.core.io.transport.mqtt'...
08:12:00.082 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/string-channel-config.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.082 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/switch-channel-config.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.098 [OH-file-processing-3] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/networkinterface.xml' in module 'org.openhab.core'...
08:12:00.098 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/rollershutter-channel-config.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.113 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/color-channel-config.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.113 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/dimmer-channel-config.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.129 [main] DEBUG org.eclipse.smarthome.core.thing.internal.firmware.FirmwareUpdateServiceImpl - Modifying the configuration of the firmware update service.
08:12:00.129 [main] DEBUG org.eclipse.smarthome.core.thing.internal.firmware.FirmwareUpdateServiceImpl - Given time unit null is not supported. Will keep current configuration.
08:12:00.145 [OH-file-processing-2] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/config.xml' in module 'org.openhab.core.thing'...
08:12:00.160 [OH-file-processing-2] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/offsetProfile.xml' in module 'org.openhab.core.thing'...
08:12:00.176 [main] DEBUG org.eclipse.smarthome.core.transform.internal.TransformationActivator - Transformation Service has been started.
08:12:00.207 [OH-file-processing-4] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/thing/thing-types.xml' in module 'org.openhab.binding.mqtt'...
08:12:00.207 [OH-file-processing-5] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/thing/homeassistant-thing.xml' in module 'org.openhab.binding.mqtt.homeassistant'...
08:12:00.207 [OH-file-processing-1] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/thing/generic-thing.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.223 [OH-file-processing-1] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/thing/channels.xml' in module 'org.openhab.binding.mqtt.generic'...
08:12:00.223 [main] INFO org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - Broker persistence file: mqttembedded.bin
08:12:00.223 [main] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - Broker anonymous access enabled
08:12:00.223 [main] DEBUG io.moquette.broker.Server - Configuring Using persistent store file, path: D:\dev\openhab2\git\openhab2-addons\itests\org.openhab.binding.mqtt.homeassistant.tests\target\test\tmp\testing\itest\userdata\mqttembedded.bin
08:12:00.223 [main] INFO io.moquette.broker.Server - Configuring message interceptors...
08:12:00.223 [main] INFO io.moquette.interception.BrokerInterceptor - Initializing broker interceptor. InterceptorIds=[]
08:12:00.223 [main] DEBUG io.moquette.broker.Server - Initialized MQTT protocol processor
08:12:00.223 [main] INFO io.moquette.broker.Server - Using default SSL context creator
08:12:00.223 [main] DEBUG io.moquette.broker.Server - Configuring MQTT authenticator
08:12:00.238 [main] INFO io.moquette.broker.Server - An io.moquette.broker.security.AcceptAllAuthenticator authenticator instance will be used
08:12:00.238 [main] DEBUG io.moquette.broker.Server - Configuring MQTT authorizator policy
08:12:00.238 [main] INFO io.moquette.broker.Server - Authorizator policy io.moquette.broker.security.PermitAllAuthorizatorPolicy instance will be used
08:12:00.238 [main] INFO io.moquette.persistence.H2Builder - Initializing H2 store
08:12:00.254 [main] INFO io.moquette.broker.subscriptions.CTrieSubscriptionDirectory - Initializing CTrie
08:12:00.254 [main] INFO io.moquette.broker.subscriptions.CTrieSubscriptionDirectory - Initializing subscriptions store...
08:12:00.254 [main] DEBUG io.moquette.persistence.H2SubscriptionsRepository - Retrieving existing subscriptions
08:12:00.254 [main] DEBUG io.moquette.persistence.H2SubscriptionsRepository - Loaded 0 subscriptions
08:12:00.301 [main] DEBUG io.netty.util.internal.logging.InternalLoggerFactory - Using SLF4J as the default logging framework
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent - Platform: Windows
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent0 - -Dio.netty.noUnsafe: false
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent0 - Java version: 8
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: unavailable
java.lang.NoClassDefFoundError: sun/misc/Unsafe
        at io.netty.util.internal.PlatformDependent0$1.run(PlatformDependent0.java:87)
        at java.security.AccessController.doPrivileged(Native Method)
        at io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:83)
        at io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:213)
        at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:81)
        at io.netty.util.ConstantPool.<init>(ConstantPool.java:32)
        at io.netty.util.AttributeKey$1.<init>(AttributeKey.java:27)
        at io.netty.util.AttributeKey.<clinit>(AttributeKey.java:27)
        at io.moquette.broker.NewNettyMQTTHandler.<clinit>(NewNettyMQTTHandler.java:37)
        at io.moquette.broker.Server.startServer(Server.java:186)
        at org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService.startEmbeddedServer(EmbeddedBrokerService.java:310)
        at org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService.initialize(EmbeddedBrokerService.java:195)
        at org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService.activate(EmbeddedBrokerService.java:159)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:228)
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41)
        at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:664)
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:510)
        at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317)
        at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:341)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:114)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:983)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:956)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:756)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:666)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:432)
        at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:665)
        at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:339)
        at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:381)
        at org.apache.felix.scr.impl.Activator.access$200(Activator.java:49)
        at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:263)
        at org.apache.felix.scr.impl.AbstractExtender.createExtension(AbstractExtender.java:196)
        at org.apache.felix.scr.impl.AbstractExtender.modifiedBundle(AbstractExtender.java:169)
        at org.apache.felix.scr.impl.AbstractExtender.modifiedBundle(AbstractExtender.java:49)
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:488)
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:1)
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:232)
        at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:450)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:911)
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:233)
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:233)
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:140)
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:132)
        at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:194)
        at org.eclipse.osgi.container.Module.publishEvent(Module.java:479)
        at org.eclipse.osgi.container.Module.start(Module.java:470)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:431)
        at aQute.launcher.Launcher.startBundles(Launcher.java:528)
        at aQute.launcher.Launcher.activate(Launcher.java:427)
        at aQute.launcher.Launcher.run(Launcher.java:306)
        at aQute.launcher.Launcher.main(Launcher.java:152)
Caused by: java.lang.ClassNotFoundException: sun.misc.Unsafe cannot be found by io.netty.common_4.1.34.Final
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:511)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:414)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:153)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 56 common frames omitted
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent0 - java.nio.DirectByteBuffer.<init>(long, int): unavailable
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.tmpdir: C:\Users\Laurent\AppData\Local\Temp (java.io.tmpdir)
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.bitMode: 64 (sun.arch.data.model)
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.maxDirectMemory: -1 bytes
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.uninitializedArrayAllocationThreshold: -1
08:12:00.301 [main] DEBUG io.netty.util.internal.CleanerJava6 - java.nio.ByteBuffer.cleaner(): available
08:12:00.301 [main] DEBUG io.netty.util.internal.PlatformDependent - -Dio.netty.noPreferDirect: false
08:12:00.316 [main] DEBUG io.moquette.broker.NewNettyAcceptor - Initializing Netty acceptor
08:12:00.316 [main] INFO io.moquette.broker.NewNettyAcceptor - Netty is using NIO
08:12:00.316 [main] DEBUG io.netty.channel.MultithreadEventLoopGroup - -Dio.netty.eventLoopThreads: 8
08:12:00.332 [main] DEBUG io.netty.channel.nio.NioEventLoop - -Dio.netty.noKeySetOptimization: false
08:12:00.332 [main] DEBUG io.netty.channel.nio.NioEventLoop - -Dio.netty.selectorAutoRebuildThreshold: 512
08:12:00.332 [main] DEBUG io.netty.util.internal.PlatformDependent - org.jctools-core.MpscChunkedArrayQueue: unavailable
08:12:00.348 [main] DEBUG io.moquette.broker.NewNettyAcceptor - Configuring TCP MQTT transport
08:12:00.348 [main] DEBUG io.moquette.broker.NewNettyAcceptor - Initializing integration. Protocol=TCP MQTT
08:12:00.348 [main] DEBUG io.moquette.broker.NewNettyAcceptor - Binding integration. host=0.0.0.0, port=1883
08:12:00.363 [main] DEBUG io.netty.channel.DefaultChannelId - -Dio.netty.processId: 15164 (auto-detected)
08:12:00.363 [main] DEBUG io.netty.util.NetUtil - -Djava.net.preferIPv4Stack: false
08:12:00.363 [main] DEBUG io.netty.util.NetUtil - -Djava.net.preferIPv6Addresses: false
08:12:00.473 [main] DEBUG io.netty.util.NetUtil - Loopback interface: lo (Software Loopback Interface 1, 127.0.0.1)
08:12:00.473 [main] DEBUG io.netty.util.NetUtil - Failed to get SOMAXCONN from sysctl and file \proc\sys\net\core\somaxconn. Default: 200
08:12:00.598 [main] DEBUG io.netty.channel.DefaultChannelId - -Dio.netty.machineId: 9c:5c:8e:ff:fe:20:3a:21 (auto-detected)
08:12:00.598 [main] DEBUG io.netty.util.internal.InternalThreadLocalMap - -Dio.netty.threadLocalMap.stringBuilder.initialSize: 1024
08:12:00.598 [main] DEBUG io.netty.util.internal.InternalThreadLocalMap - -Dio.netty.threadLocalMap.stringBuilder.maxSize: 4096
08:12:00.613 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.level: simple
08:12:00.613 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.targetRecords: 4
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.numHeapArenas: 8
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.numDirectArenas: 8
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.pageSize: 8192
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxOrder: 11
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.chunkSize: 16777216
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.tinyCacheSize: 512
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.smallCacheSize: 256
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.normalCacheSize: 64
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxCachedBufferCapacity: 32768
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.cacheTrimInterval: 8192
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.useCacheForAllThreads: true
08:12:00.629 [main] DEBUG io.netty.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxCachedByteBuffersPerChunk: 1023
08:12:00.629 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.allocator.type: pooled
08:12:00.629 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.threadLocalDirectBufferSize: 0
08:12:00.629 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.maxThreadLocalCharBufferSize: 16384
08:12:00.629 [main] INFO io.moquette.broker.NewNettyAcceptor - Server bound to host=0.0.0.0, port=1883, protocol=TCP MQTT
08:12:00.629 [main] DEBUG io.moquette.broker.NewNettyAcceptor - Configuring Websocket MQTT transport
08:12:00.644 [main] INFO io.moquette.broker.NewNettyAcceptor - Property websocket_port has been setted to disabled. Websocket MQTT will be disabled
08:12:00.644 [main] INFO io.moquette.broker.Server - Moquette integration has been started successfully in 421 ms
08:12:00.644 [main] INFO io.moquette.broker.Server - Adding MQTT message interceptor. InterceptorId=logger
08:12:00.644 [nioEventLoopGroup-2-1] DEBUG io.moquette.broker.NewNettyAcceptor - bound TCP MQTT port: 1883
08:12:00.644 [main] INFO io.moquette.interception.BrokerInterceptor - Adding MQTT message interceptor. InterceptorId=logger, handledMessageTypes=[class io.moquette.interception.messages.InterceptConnectMessage, class io.moquette.interception.messages.InterceptDisconnectMessage]
08:12:00.644 [main] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - Embedded broker connection still connecting
08:12:00.644 [nioEventLoopGroup-3-1] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0x82a5db1c, L:/127.0.0.1:1883 - R:/127.0.0.1:49443] Cid: null
08:12:00.676 [main] INFO org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection - Starting MQTT broker connection to 'localhost' with clientid embedded-mqtt-broker and file store 'D:\dev\openhab2\git\openhab2-addons\itests\org.openhab.binding.mqtt.homeassistant.tests\target\test\tmp\testing\itest\userdata\mqtt\localhost'
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxCapacityPerThread: 4096
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxSharedCapacityFactor: 2
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.linkCapacity: 16
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.ratio: 8
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.checkAccessible: true
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.checkBounds: true
08:12:00.676 [nioEventLoopGroup-3-1] DEBUG io.netty.util.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@40611b3c
08:12:00.691 [nioEventLoopGroup-3-2] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0xaa7a1967, L:/127.0.0.1:1883 - R:/127.0.0.1:49444] Cid: null
08:12:00.691 [OH-file-processing-2] DEBUG org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker - Reading the XML document '/ESH-INF/config/config.xml' in module 'org.openhab.io.mqttembeddedbroker'...
08:12:01.019 [nioEventLoopGroup-3-2] INFO io.moquette.broker.metrics.MQTTMessageLogger - C->B CONNECT <null>
08:12:01.019 [nioEventLoopGroup-3-2] DEBUG io.moquette.broker.MQTTConnection - Received MQTT message, type: CONNECT, channel: [id: 0xaa7a1967, L:/127.0.0.1:1883 - R:/127.0.0.1:49444]
08:12:01.019 [nioEventLoopGroup-3-2] DEBUG io.moquette.broker.metrics.MQTTMessageLogger - C<-B CONNACK <null>
08:12:01.035 [nioEventLoopGroup-3-2] DEBUG io.moquette.broker.MQTTConnection - Connection has been configured CId=embedded-mqtt-broker, keepAlive=60, removeTemporaryQoS2=true, idleTime=90
08:12:01.035 [nioEventLoopGroup-3-2] DEBUG io.moquette.broker.InflightResender - Initializing autoflush handler on channel [id: 0xaa7a1967, L:/127.0.0.1:1883 - R:/127.0.0.1:49444]
08:12:01.035 [nioEventLoopGroup-3-2] DEBUG io.moquette.interception.BrokerInterceptor - Sending MQTT CONNECT message to interceptor. CId=embedded-mqtt-broker, interceptorId=logger
08:12:01.035 [pool-3-thread-1] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - MQTT Client connected: embedded-mqtt-broker
08:12:01.035 [MQTT Call: embedded-mqtt-broker] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - Embedded broker connection connected
08:12:01.207 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.internal.items.ManagedMetadataProviderImpl" has been added.
08:12:01.207 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.items.ManagedItemProvider" has been added.
08:12:01.222 [main] INFO org.eclipse.smarthome.config.discovery.internal.PersistentInbox - Added new thing 'mqtt:systemBroker:embedded-mqtt-broker' to inbox.
08:12:01.222 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.ManagedThingProvider" has been added.
08:12:01.222 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.link.ManagedItemChannelLinkProvider" has been added.
08:12:01.222 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.internal.ChannelItemProvider" has been added.
08:12:01.238 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'thingManager' of size 5
08:12:01.254 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' added
08:12:01.254 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' added
08:12:01.254 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' added
08:12:01.254 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'thingLinkManager' of size 5
08:12:01.269 [main] INFO org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection - Starting MQTT broker connection to 'localhost' with clientid ha_mqtt and file store 'userdata\mqtt\localhost'
08:12:01.269 [nioEventLoopGroup-3-3] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0x88e1b6bd, L:/127.0.0.1:1883 - R:/127.0.0.1:49445] Cid: null
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedMetadataProviderImpl" has been removed.
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedItemProvider" has been removed.
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedThingProvider" has been removed.
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' removed
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.488 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' removed
08:12:01.504 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ChannelItemProvider" has been removed.
08:12:01.504 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedItemChannelLinkProvider" has been removed.
TEST reconnectTest(org.openhab.binding.mqtt.HomeAssistantMQTTImplementationTest) <<< ERROR: null
java.util.concurrent.TimeoutException
        at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1771)
        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)
        at org.openhab.binding.mqtt.HomeAssistantMQTTImplementationTest.setUp(HomeAssistantMQTTImplementationTest.java:98)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
        at junit.framework.TestSuite.runTest(TestSuite.java:252)
        at junit.framework.TestSuite.run(TestSuite.java:247)
        at aQute.junit.Activator.test(Activator.java:340)
        at aQute.junit.Activator.automatic(Activator.java:236)
        at aQute.junit.Activator.run(Activator.java:177)
        at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
        at aQute.launcher.Launcher.run(Launcher.java:352)
        at aQute.launcher.Launcher.main(Launcher.java:152)
08:12:01.504 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.internal.items.ManagedMetadataProviderImpl" has been added.
08:12:01.504 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.items.ManagedItemProvider" has been added.
08:12:01.504 [main] INFO org.eclipse.smarthome.config.discovery.internal.PersistentInbox - Added new thing 'mqtt:systemBroker:embedded-mqtt-broker' to inbox.
08:12:01.519 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.ManagedThingProvider" has been added.
08:12:01.519 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.link.ManagedItemChannelLinkProvider" has been added.
08:12:01.519 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.internal.ChannelItemProvider" has been added.
08:12:01.519 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' added
08:12:01.519 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' added
08:12:01.519 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' added
08:12:01.519 [main] INFO org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection - Starting MQTT broker connection to 'localhost' with clientid ha_mqtt and file store 'userdata\mqtt\localhost'
08:12:01.519 [nioEventLoopGroup-3-4] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0xd5f11f32, L:/127.0.0.1:1883 - R:/127.0.0.1:49446] Cid: null
08:12:01.629 [nioEventLoopGroup-3-3] INFO io.moquette.broker.metrics.MQTTMessageLogger - C->B CONNECT <null>
08:12:01.629 [nioEventLoopGroup-3-3] DEBUG io.moquette.broker.MQTTConnection - Received MQTT message, type: CONNECT, channel: [id: 0x88e1b6bd, L:/127.0.0.1:1883 - R:/127.0.0.1:49445]
08:12:01.629 [nioEventLoopGroup-3-3] DEBUG io.moquette.broker.metrics.MQTTMessageLogger - C<-B CONNACK <null>
08:12:01.629 [nioEventLoopGroup-3-3] DEBUG io.moquette.broker.MQTTConnection - Connection has been configured CId=ha_mqtt, keepAlive=60, removeTemporaryQoS2=true, idleTime=90
08:12:01.629 [nioEventLoopGroup-3-3] DEBUG io.moquette.broker.InflightResender - Initializing autoflush handler on channel [id: 0x88e1b6bd, L:/127.0.0.1:1883 - R:/127.0.0.1:49445]
08:12:01.629 [nioEventLoopGroup-3-3] DEBUG io.moquette.interception.BrokerInterceptor - Sending MQTT CONNECT message to interceptor. CId=ha_mqtt, interceptorId=logger
08:12:01.629 [nioEventLoopGroup-3-3] INFO io.moquette.broker.metrics.MQTTMessageLogger - Channel closed <ha_mqtt>
08:12:01.629 [nioEventLoopGroup-3-3] INFO io.moquette.broker.MQTTConnection - Notifying connection lost event. CId: ha_mqtt, channel: [id: 0x88e1b6bd, L:/127.0.0.1:1883 ! R:/127.0.0.1:49445]
08:12:01.644 [pool-3-thread-1] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - MQTT Client connected: ha_mqtt
08:12:01.644 [nioEventLoopGroup-3-5] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0x52da69a4, L:/127.0.0.1:1883 - R:/127.0.0.1:49447] Cid: null
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedMetadataProviderImpl" has been removed.
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedItemProvider" has been removed.
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedThingProvider" has been removed.
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' removed
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' removed
08:12:01.738 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ChannelItemProvider" has been removed.
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedItemChannelLinkProvider" has been removed.
TEST parseHATree(org.openhab.binding.mqtt.HomeAssistantMQTTImplementationTest) <<< ERROR: null
java.util.concurrent.TimeoutException
        at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1771)
        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)
        at org.openhab.binding.mqtt.HomeAssistantMQTTImplementationTest.setUp(HomeAssistantMQTTImplementationTest.java:98)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
        at junit.framework.TestSuite.runTest(TestSuite.java:252)
        at junit.framework.TestSuite.run(TestSuite.java:247)
        at aQute.junit.Activator.test(Activator.java:340)
        at aQute.junit.Activator.automatic(Activator.java:236)
        at aQute.junit.Activator.run(Activator.java:177)
        at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
        at aQute.launcher.Launcher.run(Launcher.java:352)
        at aQute.launcher.Launcher.main(Launcher.java:152)
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.internal.items.ManagedMetadataProviderImpl" has been added.
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.items.ManagedItemProvider" has been added.
08:12:01.754 [main] INFO org.eclipse.smarthome.config.discovery.internal.PersistentInbox - Added new thing 'mqtt:systemBroker:embedded-mqtt-broker' to inbox.
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.ManagedThingProvider" has been added.
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.link.ManagedItemChannelLinkProvider" has been added.
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "org.eclipse.smarthome.core.thing.internal.ChannelItemProvider" has been added.
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' added
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' added
08:12:01.754 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' added
08:12:01.769 [main] INFO org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection - Starting MQTT broker connection to 'localhost' with clientid ha_mqtt and file store 'userdata\mqtt\localhost'
08:12:01.769 [nioEventLoopGroup-3-6] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0xe867ff45, L:/127.0.0.1:1883 - R:/127.0.0.1:49448] Cid: null
08:12:01.847 [nioEventLoopGroup-3-4] INFO io.moquette.broker.metrics.MQTTMessageLogger - C->B CONNECT <null>
08:12:01.847 [nioEventLoopGroup-3-4] DEBUG io.moquette.broker.MQTTConnection - Received MQTT message, type: CONNECT, channel: [id: 0xd5f11f32, L:/127.0.0.1:1883 - R:/127.0.0.1:49446]
08:12:01.847 [nioEventLoopGroup-3-4] DEBUG io.moquette.broker.metrics.MQTTMessageLogger - C<-B CONNACK <null>
08:12:01.879 [nioEventLoopGroup-3-4] DEBUG io.moquette.broker.MQTTConnection - Connection has been configured CId=ha_mqtt, keepAlive=60, removeTemporaryQoS2=true, idleTime=90
08:12:01.879 [nioEventLoopGroup-3-4] DEBUG io.moquette.broker.InflightResender - Initializing autoflush handler on channel [id: 0xd5f11f32, L:/127.0.0.1:1883 - R:/127.0.0.1:49446]
08:12:01.879 [nioEventLoopGroup-3-4] DEBUG io.moquette.interception.BrokerInterceptor - Sending MQTT CONNECT message to interceptor. CId=ha_mqtt, interceptorId=logger
08:12:01.879 [nioEventLoopGroup-3-4] INFO io.moquette.broker.metrics.MQTTMessageLogger - Channel closed <ha_mqtt>
08:12:01.879 [nioEventLoopGroup-3-4] INFO io.moquette.broker.MQTTConnection - Notifying connection lost event. CId: ha_mqtt, channel: [id: 0xd5f11f32, L:/127.0.0.1:1883 ! R:/127.0.0.1:49446]
08:12:01.879 [pool-3-thread-1] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - MQTT Client connected: ha_mqtt
08:12:01.879 [nioEventLoopGroup-3-7] DEBUG io.moquette.broker.AutoFlushHandler - Initializing autoflush handler on channel [id: 0xb4350785, L:/127.0.0.1:1883 - R:/127.0.0.1:49449] Cid: null
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedMetadataProviderImpl" has been removed.
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedItemProvider" has been removed.
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedThingProvider" has been removed.
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' removed
08:12:01.972 [nioEventLoopGroup-3-5] INFO io.moquette.broker.metrics.MQTTMessageLogger - C->B CONNECT <null>
08:12:01.972 [nioEventLoopGroup-3-5] DEBUG io.moquette.broker.MQTTConnection - Received MQTT message, type: CONNECT, channel: [id: 0x52da69a4, L:/127.0.0.1:1883 - R:/127.0.0.1:49447]
08:12:01.972 [nioEventLoopGroup-3-5] DEBUG io.moquette.broker.metrics.MQTTMessageLogger - C<-B CONNACK <null>
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttThingHandlerFactory' removed
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.thing.internal.ThingManagerImpl - Thing handler factory 'MqttBrokerHandlerFactory' removed
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ChannelItemProvider" has been removed.
08:12:01.972 [main] DEBUG org.eclipse.smarthome.core.common.registry.AbstractRegistry - Provider "ManagedItemChannelLinkProvider" has been removed.
TEST retrieveAllTopics(org.openhab.binding.mqtt.HomeAssistantMQTTImplementationTest) <<< ERROR: null
java.util.concurrent.TimeoutException
        at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1771)
        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)
        at org.openhab.binding.mqtt.HomeAssistantMQTTImplementationTest.setUp(HomeAssistantMQTTImplementationTest.java:98)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
        at junit.framework.TestSuite.runTest(TestSuite.java:252)
        at junit.framework.TestSuite.run(TestSuite.java:247)
        at aQute.junit.Activator.test(Activator.java:340)
        at aQute.junit.Activator.automatic(Activator.java:236)
        at aQute.junit.Activator.run(Activator.java:177)
        at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
        at aQute.launcher.Launcher.run(Launcher.java:352)
        at aQute.launcher.Launcher.main(Launcher.java:152)
08:12:01.988 [nioEventLoopGroup-3-5] DEBUG io.moquette.broker.MQTTConnection - Connection has been configured CId=ha_mqtt, keepAlive=60, removeTemporaryQoS2=true, idleTime=90
08:12:01.988 [nioEventLoopGroup-3-5] DEBUG io.moquette.broker.InflightResender - Initializing autoflush handler on channel [id: 0x52da69a4, L:/127.0.0.1:1883 - R:/127.0.0.1:49447]
08:12:01.988 [nioEventLoopGroup-3-5] DEBUG io.moquette.interception.BrokerInterceptor - Sending MQTT CONNECT message to interceptor. CId=ha_mqtt, interceptorId=logger
08:12:01.988 [pool-3-thread-1] DEBUG org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService - MQTT Client connected: ha_mqtt
Tests run  : 4
Passed     : 1
Errors     : 3
Failures   : 0
[INFO] 3 Error(s)
```

--- a/itests/org.openhab.binding.mqtt.homeassistant.tests/itest.bndrun
+++ b/itests/org.openhab.binding.mqtt.homeassistant.tests/itest.bndrun
@@ -70,10 +70,12 @@ Fragment-Host: org.openhab.binding.mqtt.homeassistant
 	io.netty.handler;version='[4.1.34,4.1.35)',\
 	io.netty.resolver;version='[4.1.34,4.1.35)',\
 	io.netty.transport;version='[4.1.34,4.1.35)',\
-	org.apache.servicemix.bundles.commons-codec;version='[1.3.0,1.3.1)',\
 	org.openhab.io.mqttembeddedbroker;version='[2.5.0,2.5.1)',\
-	io.moquette.moquette-broker;version='[0.12.1,0.12.2)',\
 	tec.uom.lib.uom-lib-common;version='[1.0.3,1.0.4)',\
 	tec.uom.se;version='[1.0.10,1.0.11)',\
 	ch.qos.logback.classic;version='[1.2.3,1.2.4)',\
-	org.apache.servicemix.bundles.jaxb-impl;version='[2.2.11,2.2.12)'
+	org.apache.servicemix.bundles.jaxb-impl;version='[2.2.11,2.2.12)',\
+	moquette-broker;version='[0.13.0,0.13.1)',\
+	org.apache.commons.codec;version='[1.10.0,1.10.1)'
+
+-runvm: -Dio.netty.noUnsafe=true

--- a/itests/org.openhab.binding.mqtt.homeassistant.tests/pom.xml
+++ b/itests/org.openhab.binding.mqtt.homeassistant.tests/pom.xml
@@ -39,9 +39,9 @@
       <version>${project.version}</version>
     </dependency>
     <dependency>
-      <groupId>org.openhab.osgiify</groupId>
-      <artifactId>io.moquette.moquette-broker</artifactId>
-      <version>0.12.1</version>
+      <groupId>com.github.j-n-k</groupId>
+      <artifactId>moquette-broker</artifactId>
+      <version>0.13.0.OH2</version>
       <exclusions>
         <exclusion>
           <groupId>org.slf4j</groupId>

--- a/itests/org.openhab.binding.mqtt.homeassistant.tests/src/main/java/org/openhab/binding/mqtt/EmbeddedBrokerTools.java
+++ b/itests/org.openhab.binding.mqtt.homeassistant.tests/src/main/java/org/openhab/binding/mqtt/EmbeddedBrokerTools.java
@@ -63,7 +63,7 @@ public class EmbeddedBrokerTools {
 
             };
             mqttService.addBrokersListener(observer);
-            assertTrue("Wait for embedded connection client failed", semaphore.tryAcquire(700, TimeUnit.MILLISECONDS));
+            assertTrue("Wait for embedded connection client failed", semaphore.tryAcquire(1000, TimeUnit.MILLISECONDS));
         }
         MqttBrokerConnection embeddedConnection = this.embeddedConnection;
         if (embeddedConnection == null) {

--- a/itests/org.openhab.binding.mqtt.homeassistant.tests/src/main/java/org/openhab/binding/mqtt/HomeAssistantMQTTImplementationTest.java
+++ b/itests/org.openhab.binding.mqtt.homeassistant.tests/src/main/java/org/openhab/binding/mqtt/HomeAssistantMQTTImplementationTest.java
@@ -19,6 +19,7 @@ import static org.mockito.Mockito.*;
 import static org.mockito.MockitoAnnotations.initMocks;
 
 import java.io.IOException;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
@@ -35,6 +36,7 @@ import java.util.concurrent.TimeoutException;
 import org.eclipse.smarthome.core.library.types.OnOffType;
 import org.eclipse.smarthome.core.types.State;
 import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.core.util.UIDUtils;
 import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
 import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionObserver;
 import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionState;
@@ -92,23 +94,23 @@ public class HomeAssistantMQTTImplementationTest extends JavaOSGiTest {
 
         // Wait for the EmbeddedBrokerService internal connection to be connected
         embeddedConnection = new EmbeddedBrokerTools().waitForConnection(mqttService);
+        embeddedConnection.setRetain(true);
+        embeddedConnection.setQos(1);
 
         connection = new MqttBrokerConnection(embeddedConnection.getHost(), embeddedConnection.getPort(),
                 embeddedConnection.isSecure(), "ha_mqtt");
         connection.setQos(1);
-        connection.start().get(200, TimeUnit.MILLISECONDS);
+        connection.setRetain(true);
+        connection.setPersistencePath(Paths.get("subconn"));
+        connection.start().get(1000, TimeUnit.MILLISECONDS);
         assertThat(connection.connectionState(), is(MqttConnectionState.CONNECTED));
 
         // If the connection state changes in between -> fail
         connection.addConnectionObserver(failIfChange);
 
-        embeddedConnection.setRetain(true);
-        embeddedConnection.setQos(1);
-
         // Create topic string and config for one example HA component (a Switch)
         testObjectTopic = "homeassistant/switch/node/" + ThingChannelConstants.testHomeAssistantThing.getId();
-        final String config = "{'name':'testname','state_topic':'" + testObjectTopic + "/state','command_topic':'"
-                + testObjectTopic + "/set'}";
+        final String config = "{'name':'testname','state_topic':'" + testObjectTopic + "/state','command_topic':'" + testObjectTopic + "/set'}";
 
         // Publish component configurations and component states to MQTT
         List<CompletableFuture<Boolean>> futures = new ArrayList<>();
@@ -116,7 +118,7 @@ public class HomeAssistantMQTTImplementationTest extends JavaOSGiTest {
         futures.add(embeddedConnection.publish(testObjectTopic + "/state", "true".getBytes()));
 
         registeredTopics = futures.size();
-        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get(200, TimeUnit.MILLISECONDS);
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get(1000, TimeUnit.MILLISECONDS);
 
         failure = null;
 
@@ -144,9 +146,9 @@ public class HomeAssistantMQTTImplementationTest extends JavaOSGiTest {
     public void retrieveAllTopics() throws InterruptedException, ExecutionException, TimeoutException {
         CountDownLatch c = new CountDownLatch(registeredTopics);
         connection.subscribe("homeassistant/+/+/" + ThingChannelConstants.testHomeAssistantThing.getId() + "/#",
-                (topic, payload) -> c.countDown()).get(200, TimeUnit.MILLISECONDS);
+                (topic, payload) -> c.countDown()).get(1000, TimeUnit.MILLISECONDS);
         assertTrue("Connection " + connection.getClientId() + " not retrieving all topics",
-                c.await(200, TimeUnit.MILLISECONDS));
+                c.await(1000, TimeUnit.MILLISECONDS));
     }
 
     @Test
@@ -157,8 +159,9 @@ public class HomeAssistantMQTTImplementationTest extends JavaOSGiTest {
         Gson gson = new GsonBuilder().registerTypeAdapterFactory(new ChannelConfigurationTypeAdapterFactory()).create();
 
         ScheduledExecutorService scheduler = new ScheduledThreadPoolExecutor(4);
-        DiscoverComponents discover = spy(new DiscoverComponents(ThingChannelConstants.testHomeAssistantThing,
-                scheduler, channelStateUpdateListener, gson, transformationServiceProvider));
+        DiscoverComponents discover = spy(
+                new DiscoverComponents(ThingChannelConstants.testHomeAssistantThing, scheduler,
+                        channelStateUpdateListener, gson, transformationServiceProvider));
 
         // The DiscoverComponents object calls ComponentDiscovered callbacks.
         // In the following implementation we add the found component to the `haComponents` map
@@ -172,40 +175,41 @@ public class HomeAssistantMQTTImplementationTest extends JavaOSGiTest {
         };
 
         // Start the discovery for 100ms. Forced timeout after 300ms.
-        HaID haID = new HaID(testObjectTopic);
-        CompletableFuture<Void> future = discover.startDiscovery(connection, 100, Collections.singleton(haID), cd)
+        HaID haID = new HaID(testObjectTopic + "/config");
+        CompletableFuture<Void> future = discover.startDiscovery(connection, 500, Collections.singleton(haID), cd)
                 .thenRun(() -> {
                 }).exceptionally(e -> {
                     failure = e;
                     return null;
                 });
 
-        assertTrue(latch.await(300, TimeUnit.MILLISECONDS));
-        future.get(100, TimeUnit.MILLISECONDS);
+        assertTrue(latch.await(1000, TimeUnit.MILLISECONDS));
+        future.get(800, TimeUnit.MILLISECONDS);
 
         // No failure expected and one discovered result
         assertNull(failure);
         assertThat(haComponents.size(), is(1));
 
         // For the switch component we should have one channel group type and one channel type
-        verify(channelTypeProvider, times(1)).setChannelGroupType(any(), any());
+        // setChannelGroupType is called once above
+        verify(channelTypeProvider, times(2)).setChannelGroupType(any(), any());
         verify(channelTypeProvider, times(1)).setChannelType(any(), any());
 
-        String channelGroupId = ThingChannelConstants.testHomeAssistantThing.getId() + "_switch";
+        String channelGroupId = UIDUtils
+                .encode("node_" + ThingChannelConstants.testHomeAssistantThing.getId() + "_switch");
 
-        // We expect a switch component with an OnOff channel with the initial value UNDEF:
         State value = haComponents.get(channelGroupId).channelTypes().get(ComponentSwitch.switchChannelID).getState()
                 .getCache().getChannelState();
         assertThat(value, is(UnDefType.UNDEF));
 
         haComponents.values().stream().map(e -> e.start(connection, scheduler, 100))
                 .reduce(CompletableFuture.completedFuture(null), (a, v) -> a.thenCompose(b -> v)).exceptionally(e -> {
-                    failure = e;
-                    return null;
-                }).get();
+            failure = e;
+            return null;
+        }).get();
 
         // We should have received the retained value, while subscribing to the channels MQTT state topic.
-        verify(channelStateUpdateListener, times(1)).updateChannelState(any(), any());
+        verify(channelStateUpdateListener, timeout(1000).times(1)).updateChannelState(any(), any());
 
         // Value should be ON now.
         value = haComponents.get(channelGroupId).channelTypes().get(ComponentSwitch.switchChannelID).getState()

